cloudsmith/data_source_package.go

```diff
@@ -0,0 +1,191 @@
+package cloudsmith
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path"
+
+	"net/http"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func dataSourcePackageRead(d *schema.ResourceData, m interface{}) error {
+	pc := m.(*providerConfig)
+	namespace := requiredString(d, "namespace")
+	repository := requiredString(d, "repository")
+	identifier := requiredString(d, "identifier")
+	download := requiredBool(d, "download")
+	outputPath := requiredString(d, "output_path")
+
+	req := pc.APIClient.PackagesApi.PackagesRead(pc.Auth, namespace, repository, identifier)
+	pkg, _, err := pc.APIClient.PackagesApi.PackagesReadExecute(req)
+	if err != nil {
+		return err
+	}
+
+	d.Set("cdn_url", pkg.GetCdnUrl())
+	d.Set("format", pkg.GetFormat())
+	d.Set("is_sync_awaiting", pkg.GetIsSyncAwaiting())
+	d.Set("is_sync_completed", pkg.GetIsSyncCompleted())
+	d.Set("is_sync_failed", pkg.GetIsSyncFailed())
+	d.Set("is_sync_in_flight", pkg.GetIsSyncInFlight())
+	d.Set("is_sync_in_progress", pkg.GetIsSyncInProgress())
+	d.Set("name", pkg.GetName())
+	d.Set("slug", pkg.GetSlug())
+	d.Set("slug_perm", pkg.GetSlugPerm())
+	d.Set("version", pkg.GetVersion())
+
+	d.SetId(fmt.Sprintf("%s_%s_%s", namespace, repository, pkg.GetSlugPerm()))
+
+	var computedOutputPath string
+
+	if download {
+		err := downloadPackage(pkg.GetCdnUrl(), outputPath, pc.GetAPIKey())
+		if err != nil {
+			return err
+		}
+
+		computedOutputPath = outputPath
+	} else {
+		computedOutputPath = pkg.GetCdnUrl()
+	}
+
+	d.Set("output_path", computedOutputPath)
```

Comment:
overriding output_path like this when it's already set is going to cause issues with Terraform complaining about inconsistency, you can see this if you add a test that uses the value

you probably need two schema variables, one to set the output dir, and then use output path to output the full path including file (whether local or remote)